paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkConnectorOptions.java

```diff
@@ -413,6 +413,84 @@ public class FlinkConnectorOptions {
                     .withDescription(
                             "Optional endInput watermark used in case of batch mode or bounded stream.");
 
+    @ExcludeFromDocumentation("Only used internally to support materialized table")
+    public static final ConfigOption<CatalogTableType> CATALOG_TABLE_TYPE =
+            key("catalog-table.type")
```

Comment:
I just want to introduce an option in CoreOptions, maybe just `'type'`.

Reply:
I'm a little confused. The concept catalog table type(More specifically, materialized table) is supposed to be exclusive to flink at the moment. Does adding it to CoreOptions mean other connectors will support it in the future?

Reply:
See https://github.com/apache/paimon/pull/4243

Reply:
Yes

Reply:
Thanks, I rebased the latest master and using type in `CoreOption` now.